server/events/template_resolver.go

```diff
@@ -0,0 +1,317 @@
+
+var (
+	planCommandTitle            = command.Plan.TitleString()
+	applyCommandTitle           = command.Apply.TitleString()
+	policyCheckCommandTitle     = command.PolicyCheck.TitleString()
+	approvePoliciesCommandTitle = command.ApprovePolicies.TitleString()
+	versionCommandTitle         = command.Version.TitleString()
+	// maxUnwrappedLines is the maximum number of lines the Terraform output
+	// can be before we wrap it in an expandable template.
+	maxUnwrappedLines = 12
+)
+
+type planSuccessData struct {
+	models.PlanSuccess
+	PlanSummary              string
+	PlanWasDeleted           bool
+	DisableApply             bool
+	EnableDiffMarkdownFormat bool
+}
+
+type policyCheckSuccessData struct {
+	models.PolicyCheckSuccess
+}
+
+type ProjectOutputType int
+
+const (
+	Failure ProjectOutputType = iota
+	Error
+	PlanSuccess
+	PolicyCheckSuccess
+	ApplySuccess
+	VersionSuccess
+)
+
+func (p ProjectOutputType) String() string {
+	switch p {
+	case Failure:
+		return "project_failure"
+	case Error:
+		return "project_err"
+	case PlanSuccess:
+		return "project_plan_success"
+	case PolicyCheckSuccess:
+		return "project_policy_check_success"
+	case ApplySuccess:
+		return "project_apply_success"
+	case VersionSuccess:
+		return "project_version_success"
+	}
+	return ""
+}
+
+// Uses template overrides and server configs to resolve template
+type TemplateResolver struct {
+	// GitlabSupportsCommonMark is true if the version of GitLab we're
+	// using supports the CommonMark markdown format.
+	// If we're not configured with a GitLab client, this will be false.
+	GitlabSupportsCommonMark bool
+	DisableMarkdownFolding   bool
+	GlobalCfg                valid.GlobalCfg
+}
+
+// Resolves templates for commands
+func (t *TemplateResolver) Resolve(common commonData, baseRepo models.Repo, numPrjResults int, numPlanSuccesses int, numPolicyCheckSuccesses int, numVersionSuccesses int) *template.Template {
+	// Build template override for this repo
+	var templateOverrides map[string]string
+	repoCfg := t.GlobalCfg.MatchingRepo(baseRepo.ID())
+	if repoCfg != nil {
+		templateOverrides = repoCfg.TemplateOverrides
+	}
+
+	var tmpl *template.Template
+	switch {
+	case common.Command == approvePoliciesCommandTitle:
+		tmpl = template.Must(template.New("").Funcs(sprig.TxtFuncMap()).Parse(approveAllProjectsTmpl))
+	case common.Command == planCommandTitle, common.Command == policyCheckCommandTitle:
+		tmpl = t.getPlanTmpl(common, baseRepo, templateOverrides, numPrjResults, numPlanSuccesses, numPolicyCheckSuccesses)
+	case common.Command == applyCommandTitle:
+		tmpl = t.getApplyTmpl(templateOverrides, numPrjResults)
+	case common.Command == versionCommandTitle:
+		tmpl = t.getVersionTmpl(templateOverrides, common, numPrjResults, numVersionSuccesses)
+	}
+
+	return tmpl
+}
+
+// Resolves templates for project commands
+func (t *TemplateResolver) ResolveProject(result command.ProjectResult, baseRepo models.Repo, common commonData) (*template.Template, interface{}) {
```

Comment:
so is this the function we'd call for the checks case?

Reply:
Yup to render each project in their own checkrun output